# Week 2 Analysis

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import re

In [17]:
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)
pd.set_option('display.float_format', lambda x: '%.2f' % x)

In [18]:
df = pd.read_pickle('players')
defense = pd.read_pickle('defense')

In [19]:
print(df.shape)
df.head()

(430, 31)


Player Team Position  Age  Games  GamesStarted  \
0          Aaron Jones  GNB       RB   26      2             2   
1        Calvin Ridley  ATL       WR   26      2             2   
2         Alvin Kamara  NOR       RB   25      2             2   
3  Christian McCaffrey  CAR       RB   24      2             2   
4       Russell Wilson  SEA       QB   32      2             2   

   CompletedPasses  PassesAttempted  PassingYds  PassingTDs  Interceptions  \
0                0                0           0           0              0   
1                0                0           0           0              0   
2                0                0           0           0              0   
3                0                0           0           0              0   
4               52               63         610           9              1   

   RushingAttempts  RushingYds RushingYdspAtt  RushingTDs  Targeted  \
0               34         234          14.80           3        14   
1                1          -1          -1.00           0        22   
2               25          95          11.00           3        17   
3               41         156          11.00           4         9   
4                8          68          13.00           0         0   

   Receptions  ReceivingYds YdspReception  ReceivingTDs  Fumbles  LostFumbles  \
0           8            78         16.50             1        1            0   
1          16           239         23.40             4        0            0   
2          14           146         14.30             1        0            0   
3           7            67         14.70             0        0            0   
4           0             0          0.00             0        1            0   

   TtlTDs  TwoPTConversions  TwoPTConversionPasses FDFantasyPts PositionRank  \
0       4                 0                      0        59.20            1   
1       4                 0                      0        55.80            1   
2       4                 0                      0        55.10            2   
3       4                 0                      0        49.80            3   
4       0                 0                      0        66.20            1   

  OverallRank Ttl_Fant Week_1 Week_2  
0           1    55.20  13.60  41.60  
1           2    47.80  24.90  22.90  
2           3    48.10  18.70  29.40  
3           4    46.20  25.40  20.80  
4           5    66.20  31.80  34.40

In [20]:
print(defense.shape)
defense.head()

(32, 25)


TEAM  Ttl_Pts_Allowed  Ttl_Offense_Plays_Allowed  \
0      Baltimore Ravens               28                        125   
1  Los Angeles Chargers               41                        137   
2   Pittsburgh Steelers               43                        136   
3     Arizona Cardinals               43                        121   
4      Los Angeles Rams               43                        138   

  Yds_p_Play  Ttl_Yds  Rushing_Att  Rushing_Yds Rushing_Yds_p_Att  \
0       4.90      610           44          189              4.30   
1       5.20      709           50          247              4.90   
2       4.50      610           46          133              2.90   
3       5.60      682           48          240              5.00   
4       5.40      743           53          257              4.80   

   Rushing_TDs  Passing_Att Passing_Yds_p_Att  Completions Yds_p_Completion  \
0            0           75              6.20           46            10.10   
1            1           83              6.00           50             9.90   
2            0           80              6.90           45            12.30   
3            1           66              7.30           38            12.70   
4            3           82              6.20           51            10.00   

   Passing_Yds  Passing_TDs  RZ_Att  RZ_TD RZ_Percent  Ttl_Turnovers  \
0          464            2       3      2      66.70              5   
1          495            2       3      1      33.30              2   
2          555            4       7      2      28.60              4   
3          482            3       7      2      28.60              2   
4          508            1       7      4      57.10              3   

   Interceptions  Fumbles  Sacks Ttl_Fant Week_1 Week_2  
0              2        3      6    30.00  15.00  15.00  
1              1        1      4    12.00  11.00   1.00  
2              3        1     10    21.00   8.00  13.00  
3              0        2      7    13.00   4.00   9.00  
4              2        1      3    11.00   4.00   7.00

In [28]:
no_games = 2

defense['PtspGame'] = defense['Ttl_Pts_Allowed']/no_games
defense['YdspGame'] = defense['Ttl_Yds']/no_games
defense['RushYdspGame'] = defense['Rushing_Yds']/no_games
defense['PassYdspGame'] = defense['Passing_Yds']/no_games

## QB Analysis

In [21]:
quarterbacks = df[df['Position'] == 'QB']
quarterbacks['AttpGame'] = quarterbacks['PassesAttempted']/quarterbacks['Games']
quarterbacks['CompletionspGame'] = quarterbacks['CompletedPasses']/quarterbacks['Games']
quarterbacks['YdspGame'] = quarterbacks['PassingYds']/quarterbacks['Games']
quarterbacks['TDspGame'] = quarterbacks['PassingYds']/quarterbacks['Games']
quarterbacks.head()

/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  
/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  This is separate from the ipykernel package so we can avoid doing imports until
/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = v

Player Team Position  Age  Games  GamesStarted  CompletedPasses  \
4   Russell Wilson  SEA       QB   32      2             2               52   
6       Josh Allen  BUF       QB   24      2             2               57   
8       Cam Newton  NWE       QB   31      2             2               45   
9     Kyler Murray  ARI       QB   23      2             2               52   
11    Dak Prescott  DAL       QB   27      2             2               59   

    PassesAttempted  PassingYds  PassingTDs  Interceptions  RushingAttempts  \
4                63         610           9              1                8   
6                81         729           6              0               18   
8                63         552           1              1               26   
9                78         516           2              2               21   
11               86         716           2              0                8   

    RushingYds RushingYdspAtt  RushingTDs  Targeted  Receptions  ReceivingYds  \
4           68          13.00           0         0           0             0   
6           75           5.70           1         0           0             0   
8          122          10.90           4         0           0             0   
9          158          12.20           3         0           0             0   
11          48           6.00           3         0           0             0   

   YdspReception  ReceivingTDs  Fumbles  LostFumbles  TtlTDs  \
4           0.00             0        1            0       0   
6           0.00             0        2            2       1   
8           0.00             0        0            0       4   
9           0.00             0        0            0       3   
11          0.00             0        1            1       3   

    TwoPTConversions  TwoPTConversionPasses FDFantasyPts PositionRank  \
4                  0                      0        66.20            1   
6                  0                      0        62.70            2   
8                  0                      0        61.30            3   
9                  0                      0        60.40            4   
11                 0                      0        57.40            5   

   OverallRank Ttl_Fant Week_1 Week_2  AttpGame  CompletionspGame  YdspGame  \
4            5    66.20  31.80  34.40     31.50             26.00    305.00   
6            7    62.70  28.20  34.50     40.50             28.50    364.50   
8            9    61.30  25.70  35.60     31.50             22.50    276.00   
9           10    60.40  27.30  33.10     39.00             26.00    258.00   
11          12    57.40  17.60  39.80     43.00             29.50    358.00   

    TDspGame  
4     305.00  
6     364.50  
8     276.00  
9     258.00  
11    358.00

In [23]:
quarterbacks[quarterbacks['Player'].isin(['Dak Prescott', 'Aaron Rodgers'])]

Player Team Position  Age  Games  GamesStarted  CompletedPasses  \
11   Dak Prescott  DAL       QB   27      2             2               59   
31  Aaron Rodgers  GNB       QB   37      2             2               50   

    PassesAttempted  PassingYds  PassingTDs  Interceptions  RushingAttempts  \
11               86         716           2              0                8   
31               74         604           6              0                3   

    RushingYds RushingYdspAtt  RushingTDs  Targeted  Receptions  ReceivingYds  \
11          48           6.00           3         0           0             0   
31          14          10.70           0         1           1            -6   

   YdspReception  ReceivingTDs  Fumbles  LostFumbles  TtlTDs  \
11          0.00             0        1            1       3   
31         -6.00             0        0            0       0   

    TwoPTConversions  TwoPTConversionPasses FDFantasyPts PositionRank  \
11                 0                      0        57.40            5   
31                 0                      0        49.50            7   

   OverallRank Ttl_Fant Week_1 Week_2  AttpGame  CompletionspGame  YdspGame  \
11          12    57.40  17.60  39.80     43.00             29.50    358.00   
31          32    49.00  30.80  18.20     37.00             25.00    302.00   

    TDspGame  
11    358.00  
31    302.00

In Week 3, Dallas will be playing Seattle where Green Bay will go up against the New Orleans Saints. Both are formidable opponents so I want to hedege my bets on the defense that has allowed more passing yards & yards per passing play.

In [30]:
defense[defense['TEAM'].isin(['Seattle Seahawks', 'New Orleans Saints'])]

TEAM  Ttl_Pts_Allowed  Ttl_Offense_Plays_Allowed Yds_p_Play  \
19    Seattle Seahawks               65                        147       6.60   
21  New Orleans Saints               66                        141       4.90   

    Ttl_Yds  Rushing_Att  Rushing_Yds Rushing_Yds_p_Att  Rushing_TDs  \
19      970           46          139              3.00            3   
21      685           61          202              3.30            2   

    Passing_Att Passing_Yds_p_Att  Completions Yds_p_Completion  Passing_Yds  \
19           98              8.60           67            12.60          847   
21           74              7.00           51            10.20          521   

    Passing_TDs  RZ_Att  RZ_TD RZ_Percent  Ttl_Turnovers  Interceptions  \
19            3       9      6      66.70              3              2   
21            5       7      6      85.70              4              2   

    Fumbles  Sacks Ttl_Fant Week_1 Week_2  PtspGame  YdspGame  RushYdspGame  \
19        1      3     9.00   6.00   3.00     32.50    485.00         69.50   
21        2      6    19.00  15.00   4.00     33.00    342.50        101.00   

    PassYdspGame  
19        423.50  
21        260.50

This one is a no brainer - Seattle has allowed on average 424 passing yards so far this season where the Saints are only letting 261 yards per game.